### DISCUSSION TITLE
es/integrations/neural-magic/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#14640] [Docs]
# es/integrations/neural-magic/

Mejora el rendimiento de YOLOv8 utilizando el motor DeepSparse de Neural Magic. Aprende a desplegar y comparar modelos YOLOv8 en CPUs para una detección eficaz de objetos.

https://docs.ultralytics.com/es/integrations/neural-magic/

**Comment by wisrovi** [comment]
# Install the required packages
pip install deepsparse[yolov8]

error:
no matches found: deepsparse[yolov8]

# contexto
tengo un modelo que he exportado a onnx segun la guia: https://docs.ultralytics.com/es/integrations/onnx/#installation

luego, como la instalacion me fallo segun el error anterior, intente: pip install deepsparse, pero claramente estuvo mal, porque cuando intento importar el modelo:

from deepsparse import Pipeline
yolo_pipeline = Pipeline.create(
    task="yolov8", model_path="mymodel.onnx"
)

me da error :
2024-07-23 21:53:45 deepsparse.pipeline WARNING  Could not create v2 'yolov8' pipeline, trying legacy
File "/home/myuser/anaconda3/lib/python3.11/site-packages/deepsparse/tasks.py", line 129, in check_register_task
    raise ValueError(
ValueError: Unknown Pipeline task yolov8. Currently supported tasks are ['codegen', 'mpt', 'code_gen', 'text_generation', 'code_generation', 'opt', 'image_classification', 'llama']
File "/home/myuser/anaconda3/lib/python3.11/site-packages/deepsparse/yolov8/pipelines.py", line 26, in <module>
    from ultralytics.yolo.utils.ops import non_max_suppression as non_max_supression_torch
ModuleNotFoundError: No module named 'ultralytics.yolo'

donde se evidencia que no instale la version correcta de deepsparse, pero no puedo instalar la versi{on recomendada, no la encuentra

estoy en ubuntu22.04 con intelI5 y una NVIDIA GeForce GTX 1650 Ti,

# solicitud
me podrian ayudar, para solucionar el error y poder usar mi modelo onnx con Neural Magic


gracias

**Reply by pderrenger** [reply]
@wisrovi gracias por tu mensaje. Parece que el problema radica en la instalación de la versión correcta de DeepSparse. Primero, asegúrate de que estás utilizando la última versión de los paquetes. Puedes intentar instalar DeepSparse con el siguiente comando:

```bash
pip install 'deepsparse[yolov8]'
```

Si el problema persiste, verifica que tu entorno de Python esté configurado correctamente y que no haya conflictos con otras dependencias. Además, asegúrate de que tu modelo YOLOv8 esté exportado correctamente a ONNX utilizando el comando:

```bash
yolo task=detect mode=export model=yolov8n.pt format=onnx opset=13
```

Para más detalles, puedes consultar la documentación de Neural Magic en [este enlace](https://docs.ultralytics.com/integrations/neural-magic/). Si necesitas más ayuda, no dudes en preguntar.

**Reply by wisrovi** [reply]
gracias, con eso lo pude instalar, ahora me estoy peleando con la conversion a ONNX porque el archivo se convierte pero no me da rsultados de inferencia cercanos a los resultados antes de la conversion, pero postee este problema en https://docs.ultralytics.com/es/integrations/onnx/#how-can-i-troubleshoot-issues-when-exporting-yolov8-models-to-onnx

**Reply by pderrenger** [reply]
@wisrovi gracias por tu comentario. Me alegra saber que la instalación fue exitosa. En cuanto a la conversión a ONNX, es crucial asegurarse de que todos los parámetros y configuraciones sean correctos para mantener la precisión del modelo. Te recomiendo revisar la documentación específica sobre la exportación a ONNX y los posibles problemas que pueden surgir: https://docs.ultralytics.com/es/integrations/onnx/#how-can-i-troubleshoot-issues-when-exporting-yolov8-models-to-onnx. Si el problema persiste, asegúrate de estar utilizando las versiones más recientes de los paquetes. ¡Buena suerte!
